### ISSUE TITLE
npu编译报错：error: ‘StringToDataLayout’ is not a member of ‘phi’

### ISSUE THREAD

**Issue by shirwy** [PaddlePaddle/PaddleCustomDevice#850]
按照https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/npu/README_cn.md
指导编译时报错
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc: In function ‘void custom_kernel::BatchNormKernel(const Context&, const phi::DenseTensor&, const phi::DenseTensor&, const phi::DenseTensor&, const paddle::optional<phi::DenseTensor>&, const paddle::optional<phi::DenseTensor>&, bool, float, float, const string&, bool, bool, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*)’:
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc:64:33: error: ‘StringToDataLayout’ is not a member of ‘phi’
   const auto data_layout = phi::StringToDataLayout(data_layout_str);
                                 ^~~~~~~~~~~~~~~~~~
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc:64:33: note: suggested alternative: ‘DataLayout’
   const auto data_layout = phi::StringToDataLayout(data_layout_str);
                                 ^~~~~~~~~~~~~~~~~~
                                 DataLayout
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc: In function ‘void custom_kernel::BatchNormGradKernel(const Context&, const phi::DenseTensor&, const paddle::optional<phi::DenseTensor>&, const paddle::optional<phi::DenseTensor>&, const paddle::optional<phi::DenseTensor>&, const paddle::optional<phi::DenseTensor>&, const phi::DenseTensor&, const phi::DenseTensor&, const paddle::optional<phi::DenseTensor>&, const phi::DenseTensor&, float, float, const string&, bool, bool, bool, phi::DenseTensor*, phi::DenseTensor*, phi::DenseTensor*)’:
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc:311:33: error: ‘StringToDataLayout’ is not a member of ‘phi’
   const auto data_layout = phi::StringToDataLayout(data_layout_str);
                                 ^~~~~~~~~~~~~~~~~~
/workspace/PaddleCustomDevice/backends/npu/kernels/batch_norm_kernel.cc:311:33: note: suggested alternative: ‘DataLayout’
   const auto data_layout = phi::StringToDataLayout(data_layout_str);
                                 ^~~~~~~~~~~~~~~~~~
                                 DataLayout
CMakeFiles/paddle-custom-npu.dir/build.make:276: recipe for target 'CMakeFiles/paddle-custom-npu.dir/kernels/batch_norm_kernel.cc.o' failed
make[2]: *** [CMakeFiles/paddle-custom-npu.dir/kernels/batch_norm_kernel.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:109: recipe for target 'CMakeFiles/paddle-custom-npu.dir/all' failed
make[1]: *** [CMakeFiles/paddle-custom-npu.dir/all] Error 2
Makefile:102: recipe for target 'all' failed
make: *** [all] Error 2
+ make_error=2
+ '[' 2 '!=' 0 ']'
+ echo 'Make Error Found !!!'
Make Error Found !!!
+ exit 7

**Comment by duyongtju** [comment]
我遇到了同样的问题。
CPU aarch64 架构
paddle custom divice 版本 commit ID： 15d4633401bbf1e70798436ec5c1ebbcf45998ec
npu-smi version: 23.0.rc3

**Comment by duyongtju** [comment]
<img width="1439" alt="image" src="https://github.com/PaddlePaddle/PaddleCustomDevice/assets/33488733/f9d3e58b-9487-4294-a5ea-0186dccd88b8">

**Comment by study-hard-forever** [comment]
I had the same problem following https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/npu/README_cn.md 's process. Has it been solved？Is it the GCC and G++ compiler version? What version do you have? Mine is 7.3.0, and I see here https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/hardware_support/npu_docs/paddle_install_cn.html that version 8.2+ is needed, is it related to this version? Is this the cause?

**Comment by qili93** [comment]
this issue has been fixed by PR https://github.com/PaddlePaddle/PaddleCustomDevice/pull/853, please try agian with latest code.
